### PR TITLE
doc:Add doc for ad2s1210

### PR DESCRIPTION
**Pull request by D-Disha**
Added documentation for ad2s1210 and updated TOC

### CODE REVIEW

**Comment by D-Disha** [comment]
Hi all, are there any comments on this PR?

File: doc/sphinx/source/projects/ad2s1210/ad2s1210.rst [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+Hardware Connections
+====================
+
+Required: SDP-K1, ADZS-BRKOUT, EVAL-AD2S1210 board and USB cable.
+Plug in the EVAL-AD2S1210 board to the ADZS-BRKOUT, populate flywires to the
+SDP-K1 board using the Arduino connector.
+
+The connections are as follows:
+
+SDPK1			   BREAKOUT
```
Comment (jansunil) [inline doc/sphinx/source/projects/ad2s1210/ad2s1210.rst:54]: Should we populate the contents in a table?

File: doc/sphinx/source/projects/ad2s1210/ad2s1210.rst [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+DIGITAL-12	-	110 SDO
+DIGITAL-13	-	13  SCLK
+POWER-GND	-	4   GIO
+POWER-3v3	-	116 VIO
+
+2S1210-EVAL
+J3 CS - J3 GND
+
+Connect the SDP-K1 board to the PC using the USB cable and the AD2S1210 EVB to the
+provided AC adapter.
```
Comment (jansunil) [inline doc/sphinx/source/projects/ad2s1210/ad2s1210.rst:71]: Don't we have an interface diagram for this?

Comment (D-Disha) [reply doc/sphinx/source/projects/ad2s1210/ad2s1210.rst:71]: No this was a documentation created by Baylibre, Michael suggested we just include the minimal documentation present in our doctree

**Comment by D-Disha** [comment]
Hi all are there any more comments on this PR?

File: doc/sphinx/source/projects/ad2s1210/ad2s1210.rst [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+| DIGITAL-12  | 110 SDO    |
++-------------+------------+
+| DIGITAL-13  | 13  SCLK   |
++-------------+------------+
+| POWER-GND	  | 4 GIO      |
++-------------+------------+
+| POWER-3v3	  | 116 VIO    |
++-------------+------------+
+
+2S1210-EVAL
```
Comment (jansunil) [inline doc/sphinx/source/projects/ad2s1210/ad2s1210.rst:79]: Is there a typo here?

Comment (D-Disha) [reply doc/sphinx/source/projects/ad2s1210/ad2s1210.rst:79]: @jansunil This is how it is mentioned on the documentation created by Baylibre as well: [precision-converters-firmware/projects/ad2s1210_iio at main · analogdevicesinc/precision-converters-firmware (github.com)](https://github.com/analogdevicesinc/precision-converters-firmware/tree/main/projects/ad2s1210_iio)

I don't think this should be in the table

Comment (jansunil) [reply doc/sphinx/source/projects/ad2s1210/ad2s1210.rst:79]: No, I meant the "S1210-EVAL". Should we correct the EVAL board name if there was a typo earlier?

File: doc/sphinx/source/projects/ad2s1210/ad2s1210.rst [hunk 4]
```diff
@@ -0,0 +1,105 @@
+AD2S1210 IIO Application
```
Comment (jansunil) [inline doc/sphinx/source/projects/ad2s1210/ad2s1210.rst:1]: Any reason why this file is named as "as2s1210.rst" and not "ad2s1210_iio.rst" ?
Should we maintain a common file naming style across the repo for all project specific RST files?

File: doc/sphinx/source/projects/ad2s1210/ad2s1210.rst [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+| DIGITAL-11  | 12  SDI    |
++-------------+------------+
+| DIGITAL-12  | 110 SDO    |
++-------------+------------+
+| DIGITAL-13  | 13  SCLK   |
++-------------+------------+
+| POWER-GND	  | 4 GIO      |
++-------------+------------+
+| POWER-3v3	  | 116 VIO    |
++-------------+------------+
```
Comment (jansunil) [inline doc/sphinx/source/projects/ad2s1210/ad2s1210.rst:77]: There is some issue with rendering of this table in the index.rst

File: doc/sphinx/source/projects/ad2s1210/ad2s1210.rst (hunk 4 above)
Comment (jansunil) [reply doc/sphinx/source/projects/ad2s1210/ad2s1210.rst:1]: Sorry- typo on my end. Read as "a**d**2s1210.rst" and not "ad2s1210_iio.rst" ?

File: doc/sphinx/source/projects/ad2s1210/ad2s1210.rst [hunk 6]
```diff
@@ -0,0 +1,105 @@
...
+
+2S1210-EVAL
+J3 CS - J3 GND
```
[the remark this answers is not in the document]
Comment (D-Disha) [reply doc/sphinx/source/projects/ad2s1210/ad2s1210.rst:80]: @jansunil This is how it is mentioned on the documentation created by Baylibre as well: [precision-converters-firmware/projects/ad2s1210_iio at main · analogdevicesinc/precision-converters-firmware (github.com)](https://github.com/analogdevicesinc/precision-converters-firmware/tree/main/projects/ad2s1210_iio)

I don't think this should be in the table

File: doc/sphinx/source/projects/ad2s1210/ad2s1210.rst (hunk 3 above)
Comment (D-Disha) [reply doc/sphinx/source/projects/ad2s1210/ad2s1210.rst:79]: Apologies, this reply was for the "Should this also be a part of the table?" comment

**Review by jansunil — APPROVED** [review]

**Review by ribdp — APPROVED** [review]

**Pull request merged**